ecl/hqlcpp/hqlresource.cpp

```diff
@@ -3192,9 +3198,19 @@ bool ResourcerInfo::expandRatherThanSpill(bool noteOtherSpills)
             expr = expr->queryChild(0);
             break;
         case no_filter:
-            isFiltered = true;
-            expr = expr->queryChild(0);
-            break;
+            {
+                if (isKnownLikelihood(filterLikelihood))
+                {
+                    double likelihood = queryLikelihood(expr->queryChild(1));
+                    if (isKnownLikelihood(likelihood))
+                        filterLikelihood *= likelihood;
+                    else
+                        setUnknownLikehood(filterLikelihood);
+                }
```

Comment:
else set to unknown likelihood?  Otherwise the likelihood will be 1 when it gets to the no_table, and will get expanded.